src/main.ts

```diff
@@ -0,0 +1,260 @@
+import * as core from '@actions/core'
+import axios from 'axios'
+
+const apiEndpoint = 'http://cdm-cdb.aureacentral.com/api/sendRequest'
+
+interface CdmDatabaseInfo {
+  name: string
+  details?: string|null
+  environment: string
+  product: string
+  connection: string
+  engine: string
+  options: string|null
+  owneremail?: string|null
+  ownerskype?: string|null
+  pos?: string|null
+}
+
+interface CdmResultMessage {
+  message: string
+}
+
+type CdmResult = CdmResultMessage | CdmDatabaseInfo[]
+
+interface CdmResponse {
+  isError: boolean
+  token?: string
+  message?: string
+  result?: CdmResult
+}
+
+interface KeyMap<T> {
+  [key: string]: T
+}
+
+enum CDMCommandParameters {
+  connection = 'connection',
+  product = 'product',
+  environment = 'environment',
+  engine = "engine",
+  name = 'name',
+  details = 'details',
+  options = 'options',
+  ownermail = 'ownermail',
+  ownerskype = 'ownerskype',
+  force = 'force',
+  async = 'async'
+}
+
+let auth: string
+
+abstract class CdmCommand {
+  abstract command: string
+
+  mandatoryParameters: CDMCommandParameters[] = []
+  optionalParameters: CDMCommandParameters[] = []
+
+  abstract process () : Promise<void>
+
+  getCommandParams (): KeyMap<string> {
+    const params: KeyMap<string> = {}
+
+    this.mandatoryParameters.forEach(parameterName => {
+      const input = core.getInput(parameterName, { required: true })
+      params[parameterName] = input
+    })
+
+    this.optionalParameters.forEach(parameterName => {
+      const input = core.getInput(parameterName, { required: false })
+
+      if (input === '') {
+        return
+      }
+
+      params[parameterName] = input
+    })
+
+    return params
+  }
+
+  async getCdmResult(params: KeyMap<string> = this.getCommandParams()): Promise<CdmResponse> {
+    params.command = this.command
+
+    core.debug('Parameters:')
+    for (const key in params) {
+      const value = params[key]
+
+      core.debug(`=> ${key}:${value}`)
+    }
+
+    const { data } = await axios.get<CdmResponse>(apiEndpoint, {
+      params: params,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: `Basic ${auth}`
+      }
+    })
+
+    core.debug(`CDM result: ${JSON.stringify(data)}`)
+
+    // if (data.isError === true) {
```

Comment:
Nitpick: if this is something you might want to keep because you'll reuse it at some point, it's ok, but leave a note or a //TO DO comment for the rest of people to know whether or not this piece of code can be deleted.